de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/ui/tree/ContactPopMenu.java

```diff
@@ -38,49 +42,108 @@
 
     private final ContactTreeRootNode.ContactInfo contactInfo;
 
-    public ContactPopMenu(ContactTreeRootNode.ContactInfo contactInfo) {
+    ContactPopMenu(ContactTreeRootNode.ContactInfo contactInfo) {
         this.contactInfo = contactInfo;
 
-        if (workspace == null && project == null) {
+        if (workspace == null || project == null) {
             SarosPluginContext.initComponent(this);
+
+            if (workspace == null || project == null) {
+                LOG.error("PicoContainer injection failed. Objects still not "
+                    + "present after injection.");
+
+                return;
+            }
         }
 
+
         JMenu menuShareProject = new JMenu("Work together on...");
         menuShareProject.setIcon(IconManager.SESSIONS_ICON);
 
-        if (project == null) {
-            return;
-        }
-
         ModuleManager moduleManager = ModuleManager.getInstance(project);
 
         if(moduleManager == null){
 
-            SafeDialogUtils.showError("The local module manager could not be " +
-                "found. This most likely means that you are not using " +
-                "IntelliJ IDEA or are using an unsupported version.\n" +
-                "If you are using a supported version of IntelliJ IDEA," +
-                "please contact the Saros development team. You can reach us " +
-                "by  writing to our mailing list " +
-                "(saros-devel@googlegroups.com) or by using our contact form " +
-                "(https://www.saros-project.org/contact/Website%20feedback).",
-                "Unsupported IDE!");
+            NotificationPanel.showError(MessageFormat
+                    .format(Messages.Contact_saros_message_conditional,
+                        Messages.ContactPopMenu_unsupported_ide_message_condition),
+                Messages.ContactPopMenu_unsupported_ide_title);
 
             return;
         }
 
+        List<JMenuItem> shownModules = new LinkedList<>();
+        List<String> nonCompliantModules = new LinkedList<>();
+
         for (Module module : moduleManager.getModules()) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Reduce the total number of break and continue statements in this loop to use at most one. ([squid:S135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=240&issues=AWYbBs5MvOWp7Yd63TpS&open=AWYbBs5MvOWp7Yd63TpS)

[ISSUE_KEY]: AWYbBs5MvOWp7Yd63TpS